addons/resource/static/tests/many2many_avatar_resource_tests.js

```diff
@@ -0,0 +1,186 @@
+        //      (- avatar of the resource should not be displayed)
+        //      (Currently not implemented to stay in line with m2o widget)
+        //      - a card popover should open
+        //      - No possibility to send a message to this resource as no user exists for it
+        //  - C. human resource linked to a user
+        //      - avatar of the user should be displayed
+        //      - a card popover should open
+        //      - A button to send a message to this user should be on the card
+
+        const pyEnv = await startServer();
+        this.data = {};
+
+        // A. Material resource
+        this.data.resource1Id = pyEnv["resource.resource"].create({
+            name: "Continuity testing computer",
+            resource_type: 'material',
+        });
+
+        // B. Human resource not linked to a user
+        this.data.resource2Id = pyEnv["resource.resource"].create({
+            name: "Marie",
+            resource_type: 'user',
+        });
+
+        // C. Human resource linked to a user
+        this.data.partner1Id = pyEnv["res.partner"].create({
+            name: "Pierre",
+        });
+        this.data.user1Id = pyEnv["res.users"].create({
+            name: "Pierre",
+            partner_id: this.data.partner1Id,
+        });
+        this.data.resource3Id = pyEnv["resource.resource"].create({
+            name: "Pierre",
+            resource_type: 'user',
+            user_id: this.data.user1Id,
+            email: "Pierre@odoo.test",
+            im_status: "online",
+            phone: "+32487898933",
+        });
+
+        // Creation of the record that is linked to those resources 
+        this.data.task1Id = pyEnv["resource.task"].create({
+            display_name: "Task with three resources",
+            resource_ids: [
+                this.data.resource1Id,
+                this.data.resource2Id,
+                this.data.resource3Id,
+            ],
+        });
+    },
+}, () => {
+    QUnit.test('many2many_avatar_resource widget in form view', async function (assert) {
+        this.serverData.views = {
+            "resource.task,false,form": `
+                <form string="Tasks">
+                    <field name="display_name"/>
+                    <field name="resource_ids" widget="many2many_avatar_resource"/>
+                </form>`,
+        };
+
+        const { openView } = await start({ serverData: this.serverData });
+        await openView({
+            res_model: "resource.task",
+            res_id: this.data.task1Id,
+            views: [[false, "form"]],
+        });
+
+        assert.containsN(document.body, 'img.o_m2m_avatar', 2);
+        assert.containsN(document.body, '.fa-wrench', 1);
+
+        // Second and third records in widget should display employee avatars
+        assert.strictEqual(
+            document.querySelector(".many2many_tags_avatar_field_container .o_tag img").getAttribute("data-src"),
```

Comment:
```suggestion
        const avatarImages = document.querySelectorAll(".many2many_tags_avatar_field_container .o_tag img")
        assert.strictEqual(
            avatarImages[0].getAttribute("data-src"),
```
I think it is worth it